test/unit/rules.spec.ts

```diff
@@ -46,4 +46,29 @@ describe('Resource Graph Rule', () => {
       'Id column was not returned from Azure Resource Graph'
     );
   });
+  it('can modify a query to target resource groups', () => {
+    const rule = new ResourceGraphRule({
+      name: 'test-rule',
+      query: "Resources | where type =~ 'Microsoft.Network/virtualNetworks'",
+      description: 'Intentional bad query',
+      type: RuleType.ResourceGraph,
+    });
+    const groupNames = ['group1', 'group2', 'group3'];
+    const modifiedQuery = rule.getQueryByGroups(groupNames);
+    const expectedQuery =
+      "Resources | where resourceGroup in~ ('group1', 'group2', 'group3') | where type =~ 'Microsoft.Network/virtualNetworks'";
```

Comment:
add a task to think about & evaluate importance of: Given that we're building up string and sending it out, is there any damage that could be caused if someone tries to perform an injection attack? 

Ex: what happens if I pass in a group or a rule named `group1'; <bad stuff here> //` or similar?

Could it change or break anything? Could it make an odd join & go access data that you didn't really want it to? Or query data & then send it out to a hostile web endpoint? etc. And then is that easy/hard/likely based on usage? (It's clearly not based on project scale!)

It probably requires no action item, but let's practice spotting these things & making conscious decisions around them